File: ApiChessWebApp/Spot.cs

namespace ApiChessWebApp
{
    public class Spot//Spots has pieces
    {
      
        public Cordinates Cordinates { get; set; }

      
        public Piece Piece {get; set; }
        public bool IsSpotEmpty => this.Piece == null;
        public Spot(){}
        public Spot(Cordinates c)
        {
            this.Cordinates = c;
        }

        public Spot(Cordinates c, Piece p)
        {
            this.Cordinates = c;
            this.Piece = p;
        }
    }
}

File: UiChessWebApp/src/app/Helper/classes/Spot.ts
import { Cordinates } from "./Cordinates"
import { Piece } from "./Piece"

export class Spot{
cordinates: Cordinates;
    piece : Piece;
isSpotEmpty: boolean;

/**
 *
 */
constructor(c:Cordinates ,p: Piece,ispotempty: boolean) {
    this.cordinates = c;
    this.piece = p;
    this.isSpotEmpty = ispotempty;
    
}
}

